lib/core/test/unit/Cardano/Wallet/Primitive/ModelSpec.hs

```diff
@@ -1407,3 +1372,105 @@ blockchain =
     ]
   where
     slot e s = SlotNo $ flatSlot (EpochLength 21600) (SlotId e s)
+
+prop_applyTxToUTxO_balance :: Property
+prop_applyTxToUTxO_balance =
+    forAllShrink genTx shrinkTx $ \tx ->
+    forAllShrink genUTxO shrinkUTxO $ \u ->
+        balance (applyTxToUTxO tx u)
+        === balance u
+          `TokenBundle.add`
+              balance (utxoFromTx tx)
+          `TokenBundle.difference`
+              balance (u `UTxO.restrictedBy` Set.fromList (inputs tx))
```

Comment:
Since we don't modify the generators and shrinkers in any way, perhaps we could define `Arbitrary` instances for `Tx` and `UTxO`, and simplify these properties?

For example:

```suggestion
prop_applyTxToUTxO_balance :: Tx -> UTxO -> Property
prop_applyTxToUTxO_balance tx u =   
    balance (applyTxToUTxO tx u) ===
    balance u
        `TokenBundle.add`
            balance (utxoFromTx tx)
        `TokenBundle.difference`
            balance (u `UTxO.restrictedBy` Set.fromList (inputs tx))
```